pkg/debuginfo/manager.go

```diff
@@ -430,11 +386,16 @@ func (di *Manager) uploadFile(ctx context.Context, uploadInstructions *debuginfo
 }
 
 func uploadViaGRPC(ctx context.Context, debuginfoClient debuginfopb.DebuginfoServiceClient, uploadInstructions *debuginfopb.UploadInstructions, r io.Reader) error {
+	// NewGrpcUploadClient using bufio.NewReader to avoid closing the reader.
 	_, err := parcadebuginfo.NewGrpcUploadClient(debuginfoClient).Upload(ctx, uploadInstructions, r)
 	return err
 }
 
 func uploadViaSignedURL(ctx context.Context, url string, r io.Reader, size int64) error {
+	// Client is closing the reader if the reader is also closer.
+	// We need to wrap the reader to avoid this.
+	// We want to have total control over the reader.
+	r = bufio.NewReader(r)
```

Comment:
This one was particularly annoying to track down.

Reply:
Oh wow, that's a tricky one!